packages/backend-common/src/cache/CacheClient.ts

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright 2021 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { JsonValue } from '@backstage/config';
+import { createHash } from 'crypto';
+import Keyv from 'keyv';
+
+type CacheClientArgs = {
+  client: Keyv;
+};
+
+type CacheSetOptions = {
+  /**
+   * Optional TTL in milliseconds. Defaults to the TTL provided when the client
+   * was set up (or no TTL if none are provided).
+   */
+  ttl?: number;
+};
+
+/**
+ * A pre-configured, storage agnostic cache client suitable for use by
+ * Backstage plugins.
+ */
+export interface CacheClient {
+  /**
+   * Reads data from a cache store for the given key. If no data was found,
+   * returns undefined.
+   */
+  get(key: string): Promise<JsonValue | undefined>;
+
+  /**
+   * Writes the given data to a cache store, associated with the given key. An
+   * optional TTL may also be provided, otherwise it defaults to the TTL that
+   * was provided when the client was instantiated.
+   */
+  set(
+    key: string,
+    value: JsonValue,
+    options?: CacheSetOptions,
+  ): Promise<boolean>;
+
+  /**
+   * Removes the given key from the cache store. Resolves true if the key
+   * existed, or false if not.
+   */
+  delete(key: string): Promise<boolean>;
```

Comment:
One thing i can note before leaving for the day:

Both `set` and `delete` return a `boolean` here, rather than `void`. That may be useful for some consumers, but I just want to mention that it's a commitment to put it in the API like this. Some backing stores may need to make an additional read just to serve that return value which may run into atomicity concerns.

A boolean also doesn't evolve; we can't easily add richer return values without breaking the contract. If we want to add options, e.g. to "only delete if matching this etag/generation" which is a common need, we will want the ability to differentiate between "did not exist so nothing was done", "existed and had matching etag/generation so was successfully deleted", and "existed but did not match etag/generation so nothing was done".

If we don't have a current necessary usage of these return values, I vote that we leave them as `void` for now, until we know more about what the future holds, so we can evolve more easily.

Reply:
Yeah, this makes sense; lemme update.  Pretty sure it ended up this way because of implementation details of `keyv`, but it makes sense to decouple for the reasons you outlined.  Will update.